index.js

```diff
@@ -1,99 +1,108 @@
-const usb = require('usb')
-const yargs = require('yargs')
-const fs = require('fs')
-
+const usb = require("usb")
+const yargs = require("yargs")
+const fs = require("fs")
+const chalk = require("chalk")
 
 const argv = yargs
-    .option('f', {
-        alias: 'file',
-        describe: 'output video feed to file',
-        type: 'string'
-    })
-    .option('o', {
-        alias: 'stdout',
-        describe: 'send video feed to stdout for playback. eg: node index.js -o | ffplay -',
-        type: 'boolean'
-    })
-    .option('s', {
-        alias: 'readsize',
-        describe: 'size in bytes to queue for usb bulk interface reads',
-        default: 512,
-        type: 'integer'
-    })
-    .option('q', {
-        alias: 'queuesize',
-        describe: 'number of polling usb bulk read requests to keep in flight',
-        default: 3,
-        type: 'integer'
-    })
-    .option('v', {
-        alias: 'verbose',
-        describe: 'be noisy - doesn not play well with -o',
-        type: 'boolean'
-    })
-    .help()
-    .alias('help', 'h')
-    .argv;
-
-var goggles = usb.findByIds("0x2ca3", "0x1f")
-
-if(!goggles) {
-    console.error("Goggles USB device not found. Please connect your goggles and restart the script.")
-    process.exit(1)
-}
-goggles.open()
-if(!goggles.interfaces) {
-    console.error("Couldn't open Goggles USB device")
+  .option("f", {
+    alias: "file",
+    describe: "output video feed to file",
+    type: "string",
+  })
+  .option("o", {
+    alias: "stdout",
+    describe:
+      "send video feed to stdout for playback. eg: node index.js -o | ffplay -",
+    type: "boolean",
+  })
+  .option("s", {
+    alias: "readsize",
+    describe: "size in bytes to queue for usb bulk interface reads",
+    default: 512,
+    type: "integer",
+  })
+  .option("q", {
+    alias: "queuesize",
+    describe: "number of polling usb bulk read requests to keep in flight",
+    default: 3,
+    type: "integer",
+  })
+  .option("v", {
+    alias: "verbose",
+    describe: "be noisy - doesn not play well with -o",
+    type: "boolean",
+  })
+  .help()
+  .alias("help", "h").argv
+
+let goggles = usb.findByIds("0x2ca3", "0x1f")
+
+// When goggles are attached via USB
+usb.on("attach", (goggles) => {

```

Comment:
This isn't quite how it works:
1. The event only fires for new USB devices. You'll need to handle the case where the googles are already attached
2. The event fires for any USB device that gets attached, so you'll need to check the device VID and PID

Reply:
I'll spend more time on this when I'm home for the day. Appreciate the help!